Chess/src/main.cpp

```diff
@@ -1,40 +1,105 @@
 // Chess 
 #include "Chess.h"
+#include "Game.h"
+#include <iostream>
 
 int main()
 {
-	string board = "RNBQKBNRPPPPPPPP################################pppppppprnbqkbnr"; 
-//	string board = "##########K###############################R#############r#r#####";
-	Chess a(board);
-	int codeResponse = 0;
-	string res = a.getInput();
-	while (res != "exit")
-	{
-		/* 
-		codeResponse value : 
-		Illegal movements : 
-		11 - there is not piece at the source  
-		12 - the piece in the source is piece of your opponent
-		13 - there one of your pieces at the destination 
-		21 - illegal movement of that piece 
-		31 - this movement will cause you checkmate
-
-		legal movements : 
-		41 - the last movement was legal and cause check 
-		42 - the last movement was legal, next turn 
-		*/
-
-		/**/ 
-		{ // put your code here instead that code
-			cout << "code response >> ";
-			cin >> codeResponse;
+    string board = "RNBQKBNRPPPPPPPP################################pppppppprnbqkbnr"; 
+    //std::string board = "########R###K##############################r###k########"
+
+    //std::string board = std::string( // for testing purposes
+    //    "R###Kbr#") + 
+    //    "########" + 
+    //    "########" + 
+    //    "########" + 
+    //    "########" + 
+    //    "########" + 
+    //    "########" + 
+    //    "r###k##r";
+
+
+
+    Chess a(board);
+    Game game;
+    game.initialize(board);
+    int codeResponse = 0;
+
+    std::string res = a.getInput();
+    while (res != "exit")
+    {
+        /*
+        codeResponse value :
+        Illegal movements :
+        11 - there is not piece at the source
+        12 - the piece in the source is piece of your opponent
+        13 - there one of your pieces at the destination
+        21 - illegal movement of that piece
+        31 - this movement will cause you checkmate
+
+        legal movements :
+        41 - the last movement was legal and cause check
+        42 - the last movement was legal, next turn
+        */
+
+        // Parse input
+        if (res.length() != 4) {
+            std::cout << "Invalid input format" << std::endl;
+            res = a.getInput();
+            continue;
+        }
+
+
+
+        try {
+            auto [srcPos, destPos] = Game::parseMove(res);
+            int srcCol = srcPos.first;
+            int srcRow = srcPos.second;
+            int destCol = destPos.first;
+            int destRow = destPos.second;
```

Comment:
`first` and `second` are not indicative variable names.

Reply:
First and second are how the positionals are called in a cpp pair. I am not sure i can change this